src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -2727,19 +2727,26 @@ public void IdleOperations(object state)
 
                     foreach (var databaseKvp in DatabasesLandlord.LastRecentlyUsed.ForceEnumerateInThreadSafeManner())
                     {
-                        if (CanUnloadDatabase(databaseKvp.Key, databaseKvp.Value, statistics: null, out DocumentDatabase database) == false)
-                            continue;
+                        try
+                        {
+                            if (CanUnloadDatabase(databaseKvp.Key, databaseKvp.Value, statistics: null, out DocumentDatabase database) == false)
+                                continue;
 
-                        var dbIdEtagDictionary = new Dictionary<string, long>();
-                        using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext documentsContext))
-                        using (documentsContext.OpenReadTransaction())
+                            var dbIdEtagDictionary = new Dictionary<string, long>();
+                            using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext documentsContext))
+                            using (documentsContext.OpenReadTransaction())
+                            {
+                                foreach (var kvp in DocumentsStorage.GetAllReplicatedEtags(documentsContext))
+                                    dbIdEtagDictionary[kvp.Key] = kvp.Value;
+                            }
+
+                            if (DatabasesLandlord.UnloadDirectly(databaseKvp.Key, database.PeriodicBackupRunner.GetNextIdleDatabaseActivity(database.Name)))
+                                IdleDatabases[database.Name] = dbIdEtagDictionary;
+                        }
+                        catch (OperationCanceledException)
```

Comment:
no logging here? collateral damage?

Reply:
expected, we are checking if we can unload the database but we are already unloading it (deleting, unloading, etc)